task6/rosyk_task6.py

```diff
@@ -0,0 +1,105 @@
+import sqlite3 as sql
+from utils import number_validation_input
+
+
+def numbers_sum():
+    result = 0
+    with open('numbers.txt', encoding='utf8') as file, \
+            open('sum_numbers.txt', 'w', encoding='utf8') as output:
+        for line in file:
+            try:
+                result += float(line)
+            except ValueError:
+                print('file contains something except numbers')
+                return
+        output.write(str(result))
+    print(result)
+
+
+def odd_even_write():
+    with open('even_or_odd.txt', 'w', encoding='utf8') as file:
+        number = number_validation_input()
+        file.write(f'{number} is even' if number % 2 == 0 else f'{number} is odd')
+
+
+def python_posibilities_print():
+    with open('learning_python.txt', encoding='utf8') as file:
+        text = file.readlines()
+    list(map(print, text))
+
+
+def python_change():
+    with open('learning_python.txt', encoding='utf8') as file:
+        for line in file:
+            print(line.replace('Python', 'C'))
+
+
+def guests_greeting():
+    while True:
+        name = input('Enter guest name, enter "exit" if list is done: ')
+        if name == 'exit':
+            break
+        greeting = f'Welcome, {name}. Have a nice day!'
+        with open('guest_book.txt', 'w', encoding='utf8') as file:
```

Comment:
I would suggest to open context manager outside while loop.
now you open and close i/o stream each iteration